addons/mail/models/mail_channel.py

```diff
@@ -1115,33 +1072,40 @@ def create_group(self, partners_to, default_display_mode=False, name=''):
             'channel_type': 'group',
             'default_display_mode': default_display_mode,
             'name': name,
-            'public': 'private',
         })
         channel._broadcast(partners_to)
         return channel.channel_info()[0]
 
     @api.model
     def get_mention_suggestions(self, search, limit=8):
-        """ Return 'limit'-first channels' id, name and public fields such that the name matches a
-            'search' string. Exclude channels of type chat (DM), and private channels the current
+        """ Return 'limit'-first channels' id, name, channel_type and  authorizedGroupFullName fields such that the name matches a
+            'search' string. Exclude channels of type chat (DM), and group restricted channels the current
             user isn't registered to. """
         domain = expression.AND([
                         [('name', 'ilike', search)],
-                        [('channel_type', '=', 'channel')],
                         expression.OR([
-                            [('public', '!=', 'private')],
-                            [('channel_partner_ids', 'in', [self.env.user.partner_id.id])]
-                        ])
+                            expression.AND([
+                                [('channel_type', '=', 'channel')],
+                                expression.OR([
+                                    [('group_public_id', '=', False)],
+                                    [('channel_partner_ids', 'in', [self.env.user.partner_id.id])]
+                                ])
+                            ]),
+                            expression.AND([
+                                [('channel_type', '=', 'group')],
+                                [('channel_partner_ids', 'in', [self.env.user.partner_id.id])]
+                            ]),
```

Comment:
Allowing to mention groups is a functional change, I would not do it in this PR.